apps/avatax/src/pages/api/webhooks/order-cancelled.ts

```diff
@@ -31,50 +36,64 @@ export default wrapWithLoggerContext(
 
         logger.info("Handler called with payload");
 
-        if (!payload.order) {
-          const error = new Error("Insufficient order data");
+        const cancelledOrderFromPayload = SaleorCancelledOrderEvent.create(payload);
 
-          logger.error("Insufficient order data", { error });
-          Sentry.captureException("Insufficient order data");
+        if (cancelledOrderFromPayload.isErr()) {
+          const error = cancelledOrderFromPayload.error;
 
-          return res.status(400).send("Invalid order payload");
+          switch (true) {
+            case error instanceof OrderCancelPayloadOrderError: {
+              logger.error("Insufficient order data", { error });
+              Sentry.captureException("Insufficient order data");
+
+              return res.status(400).send("Invalid order payload");
+            }
+            case error instanceof OrderCancelNoAvataxIdError: {
+              logger.warn("No AvaTax id found in order. Likely not an AvaTax order.", { error });
+              return res.status(200).send("Invalid order payload. Likely not an AvaTax order.");
+            }
+            default: {
+              logger.error("Unhandled error", { error });
+              Sentry.captureException(error);
+
+              return res.status(500).send("Unhandled error");
+            }
```

Comment:
You don't handle "ParseError" here. We can add condition for it. It will go to sentry, but we can have specific message for it, instead of "unhandled error"